cranelift/codegen/src/isa/x64/inst.isle

```diff
@@ -3205,14 +3205,14 @@
         dst))
 
 ;; Helper for creating `palignr` instructions.
-(decl x64_palignr (Xmm XmmMem u8 OperandSize) Xmm)
-(rule 0 (x64_palignr src1 src2 imm size)
+(decl x64_palignr (Xmm XmmMem u8) Xmm)
+(rule 0 (x64_palignr src1 src2 imm)
       (xmm_rm_r_imm (SseOpcode.Palignr)
                     src1
                     src2
                     imm
-                    size))
-(rule 1 (x64_palignr src1 src2 imm size)
+                    (OperandSize.Size32)))
```

Comment:
@abrown do you perhaps remember why `palignr` was always generated with `OperandSize.Size32`? I'll admit I don't fully understand how `OperandSize` maps to instructions all the time but I naively figured that it could be "constant folded" into the `palignr` constructor, but a double-check would be good.

Reply:
Ah I remember now actually, this has to do with the rex flags when encoding where a 64-bit size forces the W bit to be set and otherwise W is unset. I believe the encoding of the `palignr` instruction forces this to "unset" so for `palignr` it should always be a non-64-bit size.

Reply:
Yeah, I think that sounds right. The `OperandSize` isn't my invention and I recall having to work around it--I think things would be more clear if we hid `OperandSize` completely or alternately surfaced it as `REX.W`, which is more direct.